iis-wf/iis-wf-import/src/main/java/eu/dnetlib/iis/wf/importer/infospace/converter/MetadataConverterUtils.java

```diff
@@ -38,13 +36,17 @@ private MetadataConverterUtils() {}
     }
     
     /**
-     * Extracts year out of the date defined in yyyy-MM-dd format.
+     * Extracts year out of the date defined in yyyy-MM-dd with possible single digit month and day part.
      */
-    public static Year extractYearOrNull(String date, Logger log) {
-        try {
-            return Year.parse(date, DateTimeFormatter.ISO_LOCAL_DATE);
-        } catch (Exception e) {
-            log.warn("unsupported, non integer, format of year value: " + date);
+    public static Integer extractYearOrNull(String date, Logger log) {
```

Comment:
I think you could do this change two ways:
1. keep `Year` as return value and change only the parsing code part
```java
    public static Year extractYearOrNull(String date, Logger log) {
        try {
            return Year.parse(date.substring(0,  date.indexOf('-')), DateTimeFormatter.ofPattern("yyyy"));
        } catch (DateTimeParseException e) {
            log.warn("unsupported, non integer, format of year value: " + date);
            return null;
        }
    }
``` 
I think working with `Year` is better than working with integer later.
2. change the return to `Integer` and do not check for blank or index of  `-`: the only check that is relevant is the result of parsing, any bad format of `date` will eventually result in exception so it doesn't matter if we do the any checks
```java
    public static Integer extractYearOrNull(String date, Logger log) {
        try {
            return Integer.valueOf(date.substring(0, date.indexOf('-')));
        } catch (DateTimeParseException e) {
            log.warn("unsupported, non integer, format of year value: " + date);
            return null;
        }
    }
```
I'd go with option number one but the second is also ok.

Reply:
Yes, there are pros and cons of either one. I'd go with 1st but with two additional updates:
1) dealing with strings without dashes to avoid `ArrayIndexOutOfBoundsException`
2) treating empty date cases differently than invalid ones (to avoid excessive logging).

One other case we could consider dealing with is simple four digit string but this case does not appear among the input data so maybe we should not introduce premature optimization.

I just pushed another commit.